apps/aecore/lib/aecore/peers/sync.ex

```diff
@@ -49,7 +49,13 @@ defmodule Aecore.Peers.Sync do
         true ->
           state
         false ->
-          Map.put(state, block_hash, block)
+          try do
+            BlockValidation.single_validate_block(block)
+            Map.put(state, block_hash, block)
+          catch
+            {:error, message} ->
+              Logger.error(fn -> "Can't add block to Sync state; #{message}" end)
```

Comment:
What happens with `state` in this case?
After `Logger.error(...)` there should be a line = `state`. So when there is an error in validation `updated_state = state`